plugins/http_plugin/http_plugin.cpp

```diff
@@ -512,23 +567,24 @@ namespace eosio {
           * @return lambda suitable for url_response_callback
           */
          template<typename T>
-         auto make_http_response_handler( detail::connection_ptr<T> con ) {
-            return [this, con]( int code, fc::variant response ) {
+         auto make_http_response_handler( detail::abstract_conn_ptr abstract_conn_ptr ) {
+            return [this, abstract_conn_ptr]( int code, fc::variant response ) {
                auto tracked_response = make_in_flight(std::move(response), *this);
-               if (!verify_max_bytes_in_flight(con)) {
+               if (!abstract_conn_ptr->verify_max_bytes_in_flight() || !abstract_conn_ptr->verify_max_requests_in_flight()) {
                   return;
                }
 
                // post  back to an HTTP thread to to allow the response handler to be called from any thread
-               boost::asio::post( thread_pool->get_executor(), [this, con, code, tracked_response=std::move(tracked_response)]() {
+               boost::asio::post( thread_pool->get_executor(), [this, abstract_conn_ptr, code, tracked_response=std::move(tracked_response)]() {
                   try {
                      std::string json = fc::json::to_string( *(*tracked_response), fc::time_point::now() + max_response_time );
                      auto tracked_json = make_in_flight(std::move(json), *this);
+                     auto con = *(static_cast<detail::connection_ptr<T>*>(*(*abstract_conn_ptr)));
```

Comment:
instead of `static_cast` here and the associated `operator*` 's above, we should move the next three lines (set_body, set_status, send) into a `virtual` method on the abstract connection that takes in  `body` and `code` and does this for us.